### PR TITLE
Ots interfaces 

### PR DESCRIPTION
**Pull request by lziq**
Applied the comment: 
https://github.com/partiql/partiql-lang-kotlin/pull/597#discussion_r883080318

### CODE REVIEW

File: lang/src/org/partiql/lang/typesystem/interfaces/operator/operators/UnaryOps.kt [hunk 1]
```diff
@@ -1,11 +1,13 @@
...
 
 import org.partiql.lang.eval.ExprValue
+import org.partiql.lang.typesystem.interfaces.operator.OpAlias
+import org.partiql.lang.typesystem.interfaces.operator.SqlOperator
 import org.partiql.lang.typesystem.interfaces.type.SqlType
 import org.partiql.lang.typesystem.interfaces.type.SqlTypeWithParameters
 import org.partiql.lang.typesystem.interfaces.type.TypeParameters
 
-abstract class UnaryOp internal constructor() : SqlOperator {
+abstract class UnaryOp(override val operatorAlias: OpAlias) : SqlOperator {
```
Comment (RCHowell) [inline lang/src/org/partiql/lang/typesystem/interfaces/operator/operators/UnaryOps.kt:10]: Change abstract to sealed

File: lang/src/org/partiql/lang/typesystem/interfaces/operator/operators/CastOps.kt [hunk 2]
```diff
@@ -1,11 +1,13 @@
...
 
 import org.partiql.lang.eval.ExprValue
+import org.partiql.lang.typesystem.interfaces.operator.OpAlias
+import org.partiql.lang.typesystem.interfaces.operator.SqlOperator
 import org.partiql.lang.typesystem.interfaces.type.SqlType
 import org.partiql.lang.typesystem.interfaces.type.SqlTypeWithParameters
 import org.partiql.lang.typesystem.interfaces.type.TypeParameters
 
-abstract class AbstractCastOp internal constructor() : SqlOperator {
+abstract class AbstractCastOp(override val operatorAlias: OpAlias) : SqlOperator {
```
Comment (RCHowell) [inline lang/src/org/partiql/lang/typesystem/interfaces/operator/operators/CastOps.kt:10]: sealed here as well

File: lang/src/org/partiql/lang/typesystem/interfaces/operator/operators/UnaryOps.kt [hunk 3]
```diff
@@ -26,3 +28,7 @@ abstract class UnaryOp internal constructor() : SqlOperator {
      */
     abstract fun invoke(sourceValue: ExprValue, typeParametersOfExpr: TypeParameters): ExprValue
 }
+
+abstract class NegOp : UnaryOp(OpAlias.NEG)
+abstract class NotOp : UnaryOp(OpAlias.NOT)
+abstract class PosOp : UnaryOp(OpAlias.POS)
```
Comment (RCHowell) [inline lang/src/org/partiql/lang/typesystem/interfaces/operator/operators/UnaryOps.kt:34]: What is the reason for these being abstract?

Comment (lziq) [reply lang/src/org/partiql/lang/typesystem/interfaces/operator/operators/UnaryOps.kt:34]: There are some methods not implemented yet in these classes. We need to inherit these classes to implement concrete operators. e.g. We can create a class `NegInt` to represent how `Neg` operator works on `Int` type.

File: lang/src/org/partiql/lang/typesystem/interfaces/operator/operators/UnaryOps.kt (hunk 1 above)
Comment (lziq) [reply lang/src/org/partiql/lang/typesystem/interfaces/operator/operators/UnaryOps.kt:10]: Will apply this comment

File: lang/src/org/partiql/lang/typesystem/interfaces/operator/operators/CastOps.kt (hunk 2 above)
Comment (lziq) [reply lang/src/org/partiql/lang/typesystem/interfaces/operator/operators/CastOps.kt:10]: Will apply this comment

**Pull request merged**